src/java/picard/vcf/FindMendelianViolations.java

```diff
@@ -0,0 +1,301 @@
+        sexChroms.addAll(FEMALE_CHROMS);
+        sexChroms.retainAll(MALE_CHROMS);
+        if (!sexChroms.isEmpty()) errors.add("The following chromosomes were listed as both male and female sex chromosomes: " + sexChroms);
+
+        // Parse the PARs
+        for (final String par : PSEUDO_AUTOSOMAL_REGIONS) {
+            try {
+                final String[] splits1 = par.split(":");
+                final String[] splits2 = splits1[1].split("-");
+                parIntervals.add(new Interval(splits1[0], Integer.parseInt(splits2[0]), Integer.parseInt(splits2[1])));
+            } catch (final Exception e) {
+                errors.add("Could not parse pseudo-autosomal region: " + par);
+            }
+        }
+
+        if (errors.isEmpty()) return null;
+        else return errors.toArray(new String[errors.size()]);
+    }
+
+    @Override
+    protected int doWork() {
+        ///////////////////////////////////////////////////////////////////////
+        // Test and then load up the inputs
+        ///////////////////////////////////////////////////////////////////////
+        final boolean outputVcfs = VCF_DIR != null;
+        IOUtil.assertFileIsReadable(INPUT);
+        IOUtil.assertFileIsReadable(TRIOS);
+        IOUtil.assertFileIsWritable(OUTPUT);
+        if (outputVcfs) IOUtil.assertDirectoryIsWritable(VCF_DIR);
+
+        LOG.info("Loading and filtering trios.");
+
+        final MendelianViolationDetector.Result result =
+                VariantProcessor.Builder
+                        .generatingAccumulatorsBy(this::buildDetector)
+                        .withInput(INPUT)
+                        .combiningResultsBy(MendelianViolationDetector.Result::merge)
+                        .multithreadingBy(THREAD_COUNT)
+                        .build()
+                        .process();
+
+        // Write out the metrics!
+        final MetricsFile<MendelianViolationMetrics, ?> metricsFile = getMetricsFile();
+        for (final MendelianViolationMetrics m : result.metrics()) {
+            m.calculateDerivedFields();
+            metricsFile.addMetric(m);
+        }
+
+        metricsFile.write(OUTPUT);
+        writeAllViolations(result);
+
+        return 0;
+    }
+
+    private void writeAllViolations(final MendelianViolationDetector.Result result) {
+        if (VCF_DIR != null) {
+            LOG.info(String.format("Writing family violation VCFs to %s...", VCF_DIR.getAbsolutePath()));
+            final VariantContextComparator vcComparator = new VariantContextComparator(inputHeader.get().getContigLines());
+            final Set<VCFHeaderLine> headerLines = new LinkedHashSet<>(inputHeader.get().getMetaDataInInputOrder());
+            headerLines.add(new VCFInfoHeaderLine("MV", 1, VCFHeaderLineType.String, "Type of mendelian violation."));
+            headerLines.add(new VCFInfoHeaderLine("AC_Orig", VCFHeaderLineCount.A, VCFHeaderLineType.Integer, "Original AC"));
```

Comment:
I'll make them public final static right here, until there's a better place. the only other place  that seems possible is HTSJDK.